### DISCUSSION TITLE
I can't validate custom Controller with TextField from material UI

### DISCUSSION THREAD

**Discussion by chanphiromsok** [react-hook-form/react-hook-form#3053] [General] [answered]
my custom component

import React from 'react'
import {useForm,Controller,Control} from 'react-hook-form';
import TextField from '@material-ui/core/TextField';
import {theme} from './them';
import {ThemeProvider} from '@material-ui/core/styles';

  

type TextField={
    name:string
    control:Control
    label: string
    type?:"text" | "number"
    style?:React.CSSProperties
    defaultValue?:unknown
    
}

export const TFCustome =({control,label,name,type="text",style,defaultValue}:TextField)=>{
    const {errors,register} = useForm();

    return(
        <div style={style}>
            <ThemeProvider theme={theme}>
        <Controller
        name={name}
        control={control}
        defaultValue={defaultValue}
        error={errors.name}
       rules={{required:true}}
        render={({onChange,name,value})=><TextField  innerRef={register({required:true})} onChange={(e)=>onChange(e)}   value={value || ''}  name={name} label={label} type={type} size="small" fullWidth variant="outlined"  />}
        />
       
        </ThemeProvider>
        </div>
    )
        
}

**Comment by bluebill1049** [comment, marked as answer]
read the doc: https://react-hook-form.com/api#Controller there is CSB as well

**Reply by chanphiromsok** [reply]
I fixed it already sir hehe it was my fault
